pkg/apis/featureflags/v0alpha1/types.go

```diff
@@ -0,0 +1,99 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/grafana/grafana/pkg/apis"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+)
+
+const (
+	GROUP      = "featureflags.grafana.app"
+	VERSION    = "v0alpha1"
+	APIVERSION = GROUP + "/" + VERSION
+)
+
+var FeatureFlagResourceInfo = apis.NewResourceInfo(GROUP, VERSION,
+	"features", "feature", "FeatureFlag",
+	func() runtime.Object { return &FeatureFlag{} },
+	func() runtime.Object { return &FeatureFlagList{} },
+)
+
+var FlagConfigResourceInfo = apis.NewResourceInfo(GROUP, VERSION,
+	"config", "config", "FlagConfig",
+	func() runtime.Object { return &FlagConfig{} },
+	func() runtime.Object { return &FlagConfigList{} },
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type FeatureFlag struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// TODO, structure so the name is not in spec
+	Spec Spec `json:"spec,omitempty"`
+}
+
+type Spec struct {
+	// Describe the feature toggle
+	Description string `json:"description"`
+
+	// How far along in development is this
+	Stage featuremgmt.FeatureFlagStage `json:"stage,omitempty"`
+
+	// Additional documentation
+	DocsURL string `json:"docsURL,omitempty"`
+
+	// Owner person or team that owns this feature flag
+	Owner string `json:"-"`
+
+	// CEL-GO expression.  Using the value "true" will mean this is on by default
+	Expression string `json:"expression,omitempty"`
+
+	// Special behavior flags
+	RequiresDevMode bool `json:"requiresDevMode,omitempty"` // can not be enabled in production
+	// This flag is currently unused.
+	RequiresRestart   bool `json:"requiresRestart,omitempty"`   // The server must be initialized with the value
+	RequiresLicense   bool `json:"requiresLicense,omitempty"`   // Must be enabled in the license
+	FrontendOnly      bool `json:"frontend,omitempty"`          // change is only seen in the frontend
+	HideFromDocs      bool `json:"hideFromDocs,omitempty"`      // don't add the values to docs
+	HideFromAdminPage bool `json:"hideFromAdminPage,omitempty"` // don't display the feature in the admin page - add a comment with the reasoning
```

Comment:
@mmandrus -- if the comment is on top, it gets added to the public openapi spec vs on the side.